users/validators.py

```diff
@@ -0,0 +1,49 @@
+# gateways/users/validators.py
+import re
+from rest_framework import serializers
+from users.models import User
+
+EMAIL_PATTERN = r"[^@]+@[^@]+\.[^@]+"
+PHONE_PATTERN = r"^\d{10,15}$"
+
+def validate_unique_email(value):
+    if User.objects.filter(email=value).exists():
+        raise serializers.ValidationError("A user with this email already exists.")
+    return value
+
+
+def validate_unique_phone(value):
+    if User.objects.filter(phone=value).exists():
+        raise serializers.ValidationError("A user with this phone number already exists.")
+    return value
+
+def validate_required_email_and_phone(attrs):
+    email = attrs.get("email")
+    phone = attrs.get("phone")
+
+    if not email:
+        raise serializers.ValidationError({"email": "Email is required."})
+    if not phone:
+        raise serializers.ValidationError({"phone": "Phone number is required."})
+    return attrs
```

Comment:
The validation should allow either of the two fields to be provided, not require both. Please adjust the logic.